dmriprep/utils/images.py

```diff
@@ -0,0 +1,178 @@
+        )
+    average_img.to_filename(out_path)
+    return out_path
+
+
+def quick_load_images(image_list, dtype=np.float32):
+    """Load 3D volumes from a list of file paths into a 4D array."""
+    example_img = nb.load(image_list[0])
+    num_images = len(image_list)
+    output_matrix = np.zeros(tuple(example_img.shape) + (num_images,), dtype=dtype)
+    for image_num, image_path in enumerate(image_list):
+        output_matrix[..., image_num] = nb.load(image_path).get_fdata(dtype=dtype)
+    return output_matrix
+
+
+def match_transforms(dwi_files, transforms, b0_indices):
+    """Arranges the order of a list of affine transforms to correspond with that of
+    each individual dwi volume file, accounting for the indices of B0s. A helper
+    function for EMC."""
+    num_dwis = len(dwi_files)
+    num_transforms = len(transforms)
+
+    if num_dwis == num_transforms:
+        return transforms
+
+    # Do sanity checks
+    if not len(transforms) == len(b0_indices):
+        raise Exception("number of transforms does not match number of b0 images")
+
+    # Create a list of which emc affines go with each of the split images
+    nearest_affines = []
+    for index in range(num_dwis):
+        nearest_b0_num = np.argmin(np.abs(index - np.array(b0_indices)))
+        this_transform = transforms[nearest_b0_num]
+        nearest_affines.append(this_transform)
+
+    return nearest_affines
+
+
+def save_4d_to_3d(in_file):
+    """Split a 4D dataset along the last dimension into multiple 3D volumes."""
+    files_3d = nb.four_to_three(nb.load(in_file))
+    out_files = []
+    for i, file_3d in enumerate(files_3d):
+        out_file = fname_presuffix(in_file, suffix="_tmp_{}".format(i))
+        file_3d.to_filename(out_file)
+        out_files.append(out_file)
+    del files_3d
+    return out_files
+
+
+def prune_b0s_from_dwis(in_files, b0_ixs):
+    """Remove *b0* volume files from a complete list of DWI volume files."""
+    if in_files[0].endswith("_warped.nii.gz"):
+        out_files = [
+            i
+            for j, i in enumerate(
+                sorted(
+                    in_files, key=lambda x: int(x.split("_")[-2].split(".nii.gz")[0])
+                )
+            )
+            if j not in b0_ixs
+        ]
+    else:
+        out_files = [
+            i
+            for j, i in enumerate(
+                sorted(
+                    in_files, key=lambda x: int(x.split("_")[-1].split(".nii.gz")[0])
+                )
+            )
+            if j not in b0_ixs
+        ]
+    return out_files
+
+
+def save_3d_to_4d(in_files):
+    """Concatenate a list of 3D volumes into a 4D output."""
+    img_4d = nb.funcs.concat_images([nb.load(img_3d) for img_3d in in_files])
+    out_file = fname_presuffix(in_files[0], suffix="_merged")
+    img_4d.to_filename(out_file)
+    del img_4d
+    return out_file
+
+
+def get_params(A):
+    """Takes an transformation affine matrix A and determines
+    rotations and translations."""
```

Comment:
Same as above for the docstring:
```suggestion
    """
    Decompose a homogeneous coordinates affine into rotation (3x3) and translations (3x1).

    Parameters
    ----------
    affine : 4x4 :py:class:`numpy.ndarray`
        A transformation affine matrix given in homogeneous coordinates.

    Returns
    -------
    rotations : 3x3 :py:class:`numpy.ndarray`
        A matrix containing the rigid-body rotations corresponding to ``affine``.
    translations : 3x1 :py:class:`numpy.ndarray`
        A vector containing the rigid-body translations component.

    """
```

Please, add some doctest `Examples` to this docstring.

Reply:
See above